ledger-service/http-json/src/main/scala/com/digitalasset/http/ConfigOps.scala

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.http
+
+import com.daml.cliopts.Logging.LogEncoder
+import com.daml.ledger.api.tls.TlsConfiguration
+import shapeless.{FieldPoly, LabelledGeneric}
+
+import java.nio.file.Path
+import scala.concurrent.duration.FiniteDuration
+import ch.qos.logback.classic.{Level => LogLevel}
+import com.daml.metrics.MetricsReporter
+import shapeless.labelled.FieldType
+import scalaz.syntax.show._
+import scalaz.std.anyVal._
+import scalaz.std.option._
+
+// This compile time checks that all fields of the Config case class
+// have a fitting implementation for displaying it on the command line.
+object ConfigOps {
+
+  // The signatures would make this code very unreadable
+  @SuppressWarnings(Array("org.wartremover.warts.ExplicitImplicitTypes"))
+  private object matcher extends FieldPoly {
+    type Field[Name, Typ] = Typ with FieldType[Symbol with shapeless.tag.Tagged[Name], Typ]
+    val W = shapeless.Witness
+
+    implicit def ledgerHost = at[Field[W.`"ledgerHost"`.T, String]](value => s"ledgerHost=$value")
```

Comment:
This needs to be that complicated because Scala 2.12 doesn't support the simple `"NiceFieldnameAsGeneric"` syntax.